lldb/unittests/Symbol/TestSwiftDemangler.cpp

```diff
@@ -69,19 +102,48 @@ TEST(TestSwiftDemangleAsyncNames, ClosureAsync) {
     EXPECT_TRUE(IsSwiftMangledName(async_name)) << async_name;
     EXPECT_TRUE(IsAnySwiftAsyncFunctionSymbol(async_name)) << async_name;
   }
+
+  CheckGroupOfFuncletsFromSameFunction(nested1_funclets);
+  CheckGroupOfFuncletsFromSameFunction(nested2_funclets1);
+  CheckGroupOfFuncletsFromSameFunction(nested2_funclets2);
+  CheckGroupOfFuncletsFromSameFunction(nested2_funclets_top_not_async);
+
+  CheckGroupOfFuncletsFromDifferentFunctions(nested1_funclets,
+                                             nested2_funclets1);
+  CheckGroupOfFuncletsFromDifferentFunctions(nested1_funclets,
+                                             nested2_funclets2);
+  CheckGroupOfFuncletsFromDifferentFunctions(nested1_funclets,
+                                             nested2_funclets_top_not_async);
+  CheckGroupOfFuncletsFromDifferentFunctions(nested2_funclets1,
+                                             nested2_funclets2);
+  CheckGroupOfFuncletsFromDifferentFunctions(nested2_funclets1,
+                                             nested2_funclets_top_not_async);
+  CheckGroupOfFuncletsFromDifferentFunctions(nested2_funclets2,
+                                             nested2_funclets_top_not_async);
```

Comment:
I know this might seem like a lot of pairwise checks, but this entire test runs in under 5ms on a _debug_ build of the binary.